### ISSUE TITLE
Transaction processing error `An account address must be 20 bytes long, got 32`

### ISSUE THREAD

**Issue by macfarla** [hyperledger/besu#4577]
stacktrace. Note `An account address must be 20 bytes long, got 32`
```
{"@timestamp":"2022-10-31T00:42:16,431","level":"INFO","thread":"vert.x-worker-thread-0","class":"AbstractBlockProcessor","message":"Block processing error: transaction invalid Internal Error in Besu - java.lang.IllegalArgumentException: An account address must be 20 bytes long, got 32. Block 0x2d4042856d73d6c82fcd7a3f1088418ed0e5eed68d1ed6c78d68d5ca89b440ad Transaction 0xea993da47eb3cc4a7d549cd5b2d52bae5b0770807f36e10aad72480d75cdd026","throwable":""}
2{"@timestamp":"2022-10-31T00:42:16,432","level":"INFO","thread":"vert.x-worker-thread-0","class":"MainnetBlockValidator","message":"Optional[Block processing error: transaction invalid Internal Error in Besu - java.lang.IllegalArgumentException: An account address must be 20 bytes long, got 32. Block 0x2d4042856d73d6c82fcd7a3f1088418ed0e5eed68d1ed6c78d68d5ca89b440ad Transaction 0xea993da47eb3cc4a7d549cd5b2d52bae5b0770807f36e10aad72480d75cdd026]. Block 2192803 (0x2d4042856d73d6c82fcd7a3f1088418ed0e5eed68d1ed6c78d68d5ca89b440ad)","throwable":""}
3{"@timestamp":"2022-10-31T00:42:16,432","level":"WARN","thread":"vert.x-worker-thread-0","class":"EngineNewPayload","message":"Invalid new payload: number: 2192803, hash: 0x2d4042856d73d6c82fcd7a3f1088418ed0e5eed68d1ed6c78d68d5ca89b440ad, parentHash: 0x7385267518859cf18f7844eb4143dfc1754f724cb158e9b8ac87932cec26b526, latestValidHash: 0x7385267518859cf18f7844eb4143dfc1754f724cb158e9b8ac87932cec26b526, status: INVALID, validationError: Block processing error: transaction invalid Internal Error in Besu - java.lang.IllegalArgumentException: An account address must be 20 bytes long, got 32. Block 0x2d4042856d73d6c82fcd7a3f1088418ed0e5eed68d1ed6c78d68d5ca89b440ad Transaction 0xea993da47eb3cc4a7d549cd5b2d52bae5b0770807f36e10aad72480d75cdd026","throwable":""}
```

**Comment by macfarla** [comment]
Could be a side effect of this change? https://github.com/hyperledger/besu/pull/4562

**Comment by shemnon** [comment]
Possibly, but I cannot tell without a stacktrace.

I expect it's an easy fix once I see the trace

**Comment by shemnon** [comment]
Stack trace was there, just the newlines and tabs were quote escaped so I didn't see the telltale indented trace.

PC alone isn't too useful.  In errors like this it was really the operation that placed that value on the stack, 13 steps earlier, that set up the error.  the `debug_traceBadBlock` RPC was what was more useful than the PC at point of failure.

**Comment by macfarla** [comment]
Introduced as a side-effect of #4562 
Attempted fix in #4578 
Both rolled back in #4583 which fixes this error.
